datagen/src/main/scala/com/nvidia/rapids/tests/scaletest/TableGenerator.scala

```diff
@@ -0,0 +1,269 @@
+      "b_key3_1 string," +
+      "b_key3_2 date," +
+      "b_key3_3 long," +
+      (1 to 10).map(i => s"b_data_$i ${randomColumnType()}").mkString(", ") + ", " +
+      (1 to 10).map(i => s"b_data_small_value_range_$i decimal(10, 2)").mkString(", ")
+    val overlapRatio = 0.99
+    val bData = dbgen.addTable("b_data", schema, bNumRows)
+    // This column "Should overlap with a_facts.primary_a about 99% of the time"
+    val overlapSeeds = Math.round(aNumRows * overlapRatio)
+    val startSeed = aNumRows - overlapSeeds + 1
+    val endSeed = startSeed + aNumRows
+    bData("b_foreign_a").setSeedRange(startSeed, endSeed)
+    bData.configureKeyGroup(
+      (1 to 3).map(i => s"b_key3_$i"),
+      CorrelatedKeyGroup(3, 1, Math.round(bNumRows * 0.99)),
+      FlatDistribution())
+    (1 to 10).map(i => bData(s"b_data_small_value_range_$i").setSeedRange(1, 10000))
+    bData.toDF(spark)
+  }
+
+  /**
+  c_data: Each scale factor corresponds to 100,000 rows.
+    - c_foreign_a Should overlap with a_facts.primary_a about 50% of the time
+    - key group 2 columns up to the complexity (each key should show up about 10 times)
+    - c_data_row_num_1 1 data column that is essentially the row number for order by tests in window
+    - 5 data columns
+    - 5 numeric data columns
+   */
+  private def genCData(): DataFrame = {
+    val schema = "c_foreign_a long," +
+      (1 to complexity).map(i => s"c_key2_$i ${randomColumnType()}").mkString(",") + "," +
+      "c_data_row_num_1 long," +
+      (1 to 5).map(i => s"c_data_$i ${randomColumnType()}").mkString(",") + "," +
+      (1 to 5).map(i => s"c_data_numeric_$i ${randomNumericColumnType()}").mkString(",")
+    val overlapRatio = 0.50
+    val cData = dbgen.addTable("c_data", schema, cNumRows)
+    val overlapSeeds = Math.round(aNumRows * overlapRatio)
+    val startSeed = aNumRows - overlapSeeds + 1
+    val endSeed = startSeed + aNumRows
+    cData("c_foreign_a").setSeedRange(startSeed, endSeed)
+    cData.configureKeyGroup(
+      (1 to complexity).map(i => s"c_key2_$i"),
+      CorrelatedKeyGroup(2, 1, cNumRows/10),
+      FlatDistribution())
+    cData("c_data_row_num_1").setValueGen(RowNumPassThrough)
+    cData.toDF(spark)
+  }
+  /**
+  d_data: Each scale factor corresponds to 100,000 rows.
+    - key group 2 complexity columns (each key should show up about 10 time, but the overlap with c_data for key group 2 should only be about 50%)
+    - 10 data columns
+   */
+  private def genDData(): DataFrame = {
+    val schema = (1 to complexity).map(i => s"d_key2_$i ${randomColumnType()}").mkString(",") +
+      "," +
+      (1 to 10).map(i => s"d_data_$i ${randomColumnType()}").mkString(",")
+    val dData = dbgen.addTable("d_data", schema, dNumRows)
+    // each key should show up about 10 time, but the overlap with c_data for key group 2 should only be about 50%
+    // c_data key group uses seed range (1, 10) so here d_data uses (6, 15) to get 50% overlap
```

Comment:
nit: please update the comment here. All you need to do is add in a "for example" at the beginning.

```
For example if c_data.c_key2_* had 100 rows the seed range would be (1, 10) so here d_data.d_key2_* would use
a seed range of (5, 15)
```

But that shows an off by one error in my calculation now that I do it manually. It is probably fine because being off by 1 is not the end of the world here.